### PR TITLE
android: update to use NDK 23c

### PR DESCRIPTION
**Pull request by benoit-pierre**
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1620)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -238,20 +242,6 @@ $(OUTPUT_DIR)/extr: extr.c $(MUPDF_LIB) $(MUPDF_DIR)/include $(JPEG_LIB) $(FREET
...
-#
-# --deprecated-headers is necessary in NDK 15, but will fail in 12-14
-android-toolchain:
-ifneq ($(wildcard $(NDK)/build/tools),)
-	$(NDK)/build/tools/make_standalone_toolchain.py --force --install-dir=$(ANDROID_TOOLCHAIN) \
-		--arch $(ANDROID_ARCH) --api $(NDKABI) --deprecated-headers || \
-	$(NDK)/build/tools/make_standalone_toolchain.py --force --install-dir=$(ANDROID_TOOLCHAIN) \
-		--arch $(ANDROID_ARCH) --api $(NDKABI)
-endif
-
```
Comment (Frenzie) [inline Makefile:254]: Please don't remove this.

Comment (benoit-pierre) [reply Makefile:254]: Newer NDKs don't support this: there's no need for it, as the toolchains can directly be used.

Comment (benoit-pierre) [reply Makefile:254]: Starting with NDK r19c: see https://github.com/android/ndk/wiki/Changelog-r19.

Comment (Frenzie) [reply Makefile:254]: In that case don't forget to remove it from .PHONY at the bottom.

Comment (benoit-pierre) [reply Makefile:254]: Done.

**Review by Frenzie — APPROVED** [review]
lgtm, can't test atm

**Comment by benoit-pierre** [comment]
Need an update to the Android docker image: https://github.com/koreader/virdevenv/pull/86.

**Comment by Frenzie** [comment]
Btw, the other builds are failing too.

```
-- FLOATTEST = no-fp-contract
-- RPM architecture = , DEB architecture = 
-- Configuring incomplete, errors occurred!
See also "/home/ko/base/thirdparty/libjpeg-turbo/build/arm-kindle5-linux-gnueabi/libjpeg-turbo-prefix/src/libjpeg-turbo-build/CMakeFiles/CMakeOutput.log".
ninja: build stopped: subcommand failed.
Makefile.third:78: recipe for target 'build/arm-kindle5-linux-gnueabi/libs/libjpeg.so.8' failed
make: *** [build/arm-kindle5-linux-gnueabi/libs/libjpeg.so.8] Error 1

Exited with code exit status 2

```

**Comment by NiLuJe** [comment]
My only comment is about the `SHARED_STL_LINK_FLAG`; I'm assuming it's no longer necessary and shit just works?

**Comment by benoit-pierre** [comment]
> My only comment is about the `SHARED_STL_LINK_FLAG`; I'm assuming it's no longer necessary and shit just works?

More or less, yes.

**Comment by benoit-pierre** [comment]
FIXME: drop `openssl-gcc-4.9-stdatomic.patch` once `sony-prstux` is gone too.

**Comment by poire-z** [comment]
All good for you @Frenzie @NiLuJe ?

**Comment by Frenzie** [comment]
Not quite yet, see https://github.com/koreader/koreader/pull/10639

**Review by Frenzie — APPROVED** [review]
lgtm

**Review by NiLuJe — CHANGES_REQUESTED** [review]
> Reviewed 6 of 12 files at r1, 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1620)* status: all files reviewed, 2 unresolved discussions (waiting on @benoit-pierre and @Frenzie)

___
*[`Makefile.defs` line 567 at r2](https://reviewable.io/reviews/koreader/koreader-base/1620#-NZXHez42emyyrqn-HwW:-NZXHez42emyyrqn-HwX:b-4ifgqz) ([raw file](https://github.com/koreader/koreader-base/blob/d2f4ce51a8bb09ea86f6706aa96c539029536590/Makefile.defs#L567)):*
> ```
> 		LDFLAGS+=-Wl,--fix-cortex-a8 -march=armv7-a
> 	endif
> 	# Fix https://github.com/android/ndk/issues/1615
> ```

That issue looks... weird?

Anything in particular force-feeding libgcc helped with in practice?

In my vague experience with Linux cross clang builds, libgcc can *usually* be dropped because we use compiler-rt and libunwind instead, and, when we don't, clang deals with it automatically by passing the right gcc-toolchain & gcc-install-dir paths. But that's on Linux ;o).

**Comment by benoit-pierre** [comment]
> > Reviewed 6 of 12 files at r1, 8 of 8 files at r2, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1620)_ status: all files reviewed, 2 unresolved discussions (waiting on @benoit-pierre and @Frenzie)
> 
> _[`Makefile.defs` line 567 at r2](https://reviewable.io/reviews/koreader/koreader-base/1620#-NZXHez42emyyrqn-HwW:-NZXHez42emyyrqn-HwX:b-4ifgqz) ([raw file](https://github.com/koreader/koreader-base/blob/d2f4ce51a8bb09ea86f6706aa96c539029536590/Makefile.defs#L567)):_
> 
> > ```
> > 		LDFLAGS+=-Wl,--fix-cortex-a8 -march=armv7-a
> > 	endif
> > 	# Fix https://github.com/android/ndk/issues/1615
> > ```
> 
> That issue looks... weird?
> 
> Anything in particular force-feeding libgcc helped with in practice?
> 
> In my vague experience with Linux cross clang builds, libgcc can _usually_ be dropped because we use compiler-rt and libunwind instead, and, when we don't, clang deals with it automatically by passing the right gcc-toolchain & gcc-install-dir paths. But that's on Linux ;o).

Off by one error! Should have been https://github.com/android/ndk/issues/1614

**Comment by benoit-pierre** [comment]
Anyway, a workaround for other missing symbol issues when loading libraries… Not necessary with the meson branch using monolibtic mode.

**Review by NiLuJe — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1620)* status: all files reviewed, 1 unresolved discussion (waiting on @Frenzie)

**Comment by poire-z** [comment]
> Not quite yet, see [koreader/koreader#10639](https://github.com/koreader/koreader/pull/10639)

So, I guess we should wait until saturday to go on merging these PRs?

**Comment by Frenzie** [comment]
Yes, it'll still be quite low on translations but this isn't a string I want to push in last minute. :-)

**Comment by poire-z** [comment]
Good to go ? @benoit-pierre : available these days/evenings to go on with all that?

**Comment by benoit-pierre** [comment]
> Good to go ? @benoit-pierre : available these days/evenings to go on with all that?

Yes.

**Pull request merged**